src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx

```diff
@@ -0,0 +1,199 @@
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Toolbar,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import FlagOutlinedIcon from "@mui/icons-material/FlagOutlined";
+import { Link, createFileRoute } from "@tanstack/react-router";
+import { FlexColumnLayout } from "components/generic/flex-column-layout";
+import LoadingTableCell from "components/generic/loading-table-cell";
+import { useFindProjectMilestoneQuery, useListMilestoneTasksQuery } from "hooks/api-queries";
+import { useTranslation } from "react-i18next";
+import { DateTime } from "luxon";
+import ProgressBadge from "components/generic/progress-badge";
+import TaskButton from "components/tasks/new-task-button";
+import TaskDialog from "components/tasks/task-dialog";
+import React from "react";
+import { Task } from "generated/client";
+
+/**
+ * Milestone tasks file route
+ */
+export const Route = createFileRoute("/projects/$projectId/schedule/$milestoneId/tasks")({
+  component: MilestoneTasksListRoute,
+});
+
+/**
+ * Milestone tasks list route component
+ */
+function MilestoneTasksListRoute() {
+  const { t } = useTranslation();
+  const { projectId, milestoneId } = Route.useParams();
+
+  const findProjectMilestoneQuery = useFindProjectMilestoneQuery(projectId, milestoneId);
+  const milestone = findProjectMilestoneQuery.data;
+  const listMilestoneTasksQuery = useListMilestoneTasksQuery({ projectId, milestoneId });
+  const tasks = listMilestoneTasksQuery.data;
+
+  const [open, setOpen] = React.useState(false);
+  const [task, setTask] = React.useState<null | Task>(null);
+
+  /**
+   * Handles task select
+   */
+  const onTaskSelect = (task: Task) => {
+    setTask(task);
+    setOpen(true);
+  };
+
+  /**
+   * Handles task close
+   */
+  const onTaskClose = () => {
+    setTask(null);
+    setOpen(false);
+  }
+
+  /**
+   * Renders the milestone tasks rows
+   */
+  const renderMilestoneTasksRows = () => {
+    if (listMilestoneTasksQuery.isFetching) {
+      return (
+        <TableRow>
+          <LoadingTableCell loading />
+          <LoadingTableCell loading />
+          <LoadingTableCell loading />
+        </TableRow>
+      );
+    }
+
+    return (tasks ?? []).map((task) => {
+      const startDate = DateTime.fromJSDate(task.startDate);
+      const endDate = DateTime.fromJSDate(task.endDate);
+      const difference = endDate.diff(startDate, "days").days;
+      const formattedStartDate = startDate.toFormat("dd.MM.yyyy");
+      const formattedEndDate = endDate.toFormat("dd.MM.yyyy");
+
+      return (
+        <TableRow key={task.id}>
+          <TableCell style={{ overflow: "hidden" }}>
+            <div style={{ display: "flex", alignItems: "center", gap: 8 }}>
+              <Avatar sx={{ backgroundColor: "#0079BF" }}>
+                <FlagOutlinedIcon fontSize="large" sx={{ color: "#fff" }} />
+              </Avatar>
+              {/* TODO: Handle overflowing name with maxWidth could be improved */}
```

Comment:
Again if you think this is fine, feel free to remove the comment

Reply:
Let's keep it for now, maybe we can verify and remove it in our future PRs